src/util.js

```diff
@@ -64,14 +65,17 @@ export function styleObjToCss(s) {
 					: JS_TO_CSS[prop] ||
 					  (JS_TO_CSS[prop] = prop.replace(CSS_REGEX, '-$1').toLowerCase());
 
-			str =
-				str +
-				name +
-				':' +
-				val +
-				(typeof val === 'number' && IS_NON_DIMENSIONAL.test(prop) === false
-					? 'px;'
-					: ';');
+			let suffix = ';';
+			if (SUFFIX_CACHE[name]) {
```

Comment:
Do we need to check if `typeof val === 'number'` here as well? If the first call to render is `<div style={{ padding: 8 }} />` but the second is `<div style={{ padding: '8px' }} />`, will it still add the `'px'`?